tests/ui/expr_metavars_in_unsafe.rs

```diff
@@ -0,0 +1,228 @@
+#![feature(decl_macro)]
+#![warn(clippy::expr_metavars_in_unsafe)]
+#![allow(clippy::no_effect)]
+
+#[macro_export]
+macro_rules! allow_works {
+    ($v:expr) => {
+        #[allow(clippy::expr_metavars_in_unsafe)]
```

Comment:
Could you also add a test for `#[expect]`? That will require the `lint_reasons` feature.